source/symbols/Type.cpp

```diff
@@ -202,7 +202,19 @@ bool Type::isStruct() const {
 
 bool Type::isBitstreamType() const {
     // TODO: classes
-    return isIntegral() || isUnpackedArray() || isUnpackedStruct();
+    if (isIntegral())
+        return true;
+    if (isUnpackedArray())
+        return getArrayElementType()->isBitstreamType();
+    if (isUnpackedStruct()) {
+        auto& us = as<UnpackedStructType>();
```

Comment:
You have to use getCanonicalType() here in order to safely use as<>. Most other Type methods handle checking for canonical types for you automatically, but as<> casting is more low level.

Reply:
Yes, I change the problematic line into
```c++
  auto& us = getCanonicalType().as<UnpackedStructType>();
```